### PR TITLE
21). Add topic files 55 to 58

### PR DESCRIPTION
**Pull request by jameshkramer**
@mairaw @BillWagner

This PR is for user story #1576037.

It adds four article files, makes corresponding deletions from the C# and VB folders, and adds redirection entries. There are also fixes to past articles.

Please merge #17732 before merging this.

cc: @tfosmark @marteeleigh

Contributes to #4728

### CODE REVIEW

**Review by BillWagner — APPROVED** [review]
I'm happy with this set of changes @jameshkramer 

I'll let @mairaw weigh in before we merge.

**Comment by mairaw** [comment]
@jameshkramer can you fix the merge conflicts here before I start the review?

File: docs/standard/linq/sort-elements-multiple-keys.md [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+  - "vb"
 ms.assetid: 3b2760b6-d607-4ac7-b784-5c6524e2a0e0
 ---
-# How to sort elements on multiple keys (C#)
 
-This topic shows how to sort on multiple keys.
+# How to sort elements on multiple keys (LINQ to XML)
 
-## Example
+This article shows how to sort on multiple keys.
```
Comment (mairaw) [inline docs/standard/linq/sort-elements-multiple-keys.md:13]: In some cases, we could add the keywords C# or Visual Basic to the intro to make sure we don't lose the SEO that used to be in the title.

```suggestion
This article shows how to sort on multiple keys in C# or Visual Basic.
```

File: docs/standard/linq/debug-empty-query-results-sets.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  - "csharp"
+  - "vb"
+ms.assetid: b569f0dc-425e-45a6-acbf-770fb761c981
+---
+
+# How to debug empty query results sets (LINQ to XML)
+
+One of the most common problems when querying XML trees is that if the XML tree has a default namespace, the developer sometimes writes the query as though the XML were not in a namespace.
+
+The first set of examples in this topic shows a typical way that XML in a default namespace is loaded, and then queried improperly.
```
Comment (mairaw) [inline docs/standard/linq/debug-empty-query-results-sets.md:15]:
```suggestion
The first set of examples in this article shows a typical way that XML in a default namespace is loaded, and then queried improperly.
```

File: docs/standard/linq/calculate-intermediate-values.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ms.date: 07/20/2015
+dev_langs:
+  - "csharp"
+  - "vb"
+ms.assetid: 7fd3001f-f8f9-4bce-879f-d4c7af8a04fe
+---
+
+# How to calculate intermediate values (LINQ to XML)
+
+This article shows how to calculate intermediate values for use in sorting, filtering, and selecting.
```
Comment (mairaw) [inline docs/standard/linq/calculate-intermediate-values.md:13]:
```suggestion
This article shows how to calculate intermediate values for use in sorting, filtering, and selecting in C# and Visual Basic.
```

**Review by mairaw — COMMENTED** [review]
A few minor comments before we merge

**Review by mairaw — APPROVED** [review]

**Pull request merged**